src/MiddlewareListener.php

```diff
@@ -75,7 +99,9 @@ public function onDispatch(MvcEvent $event)
             $event->setName(MvcEvent::EVENT_DISPATCH_ERROR);
             $event->setError($application::ERROR_EXCEPTION);
             $event->setController($middlewareName);
-            $event->setControllerClass(get_class($middleware));
+            if (null !== $middlewareToBePiped) {
+                $event->setControllerClass(get_class($middlewareToBePiped));
```

Comment:
You always assume that the last piped middleware is the "responding" one here... Probably needs documenting/commenting